master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go

```diff
@@ -0,0 +1,176 @@
+}
+
+func TestGetAgentsAndGetSlots(t *testing.T) {
+	// Create agent nodes.
+	auxNode := createMockNode(auxNodeName)
+	compNode := createMockNode(compNodeName)
+
+	nodes := make(map[string]*k8sV1.Node)
+	nodes[auxNode.Name] = &auxNode
+	nodes[compNode.Name] = &compNode
+
+	// Create resource managers with different device types for their respective pod service.
+	rmGPU := createMockRM(nodes, podServSlotTypeGPU)
+	rmCUDA := createMockRM(nodes, device.CUDA)
+	rmCPU := createMockRM(nodes, device.CPU)
+
+	rmList := []*ResourceManager{rmGPU, rmCUDA, rmCPU}
+	for _, rm := range rmList {
+		agentsResp, err := rm.GetAgents(&apiv1.GetAgentsRequest{})
+		require.NoError(t, err)
+		require.Equal(t, len(nodes), len(agentsResp.Agents))
+		for _, agent := range agentsResp.Agents {
+			_, ok := nodes[agent.Id]
+			require.True(t, ok)
+			agentResp, err := rm.GetAgent(&apiv1.GetAgentRequest{AgentId: agent.Id})
+			require.NoError(t, err)
+			require.Equal(t, agentResp.Agent.Id, agent.Id)
+
+			slotsResp, err := rm.GetSlots(&apiv1.GetSlotsRequest{AgentId: agent.Id})
+			require.NoError(t, err)
+			require.Equal(t, nodeNumSlots, len(slotsResp.Slots))
+			activeSlots := 0
+			cnt := 0
+			for _, slot := range slotsResp.Slots {
+				slotResp, err := rm.GetSlot(&apiv1.GetSlotRequest{
+					AgentId: agent.Id,
+					SlotId:  strconv.Itoa(cnt),
+				})
+				require.NoError(t, err)
+				slotID, err := strconv.Atoi(slotResp.Slot.Id)
+				require.NoError(t, err)
+				require.Equal(t, cnt, slotID)
+				if slot.Container != nil {
+					activeSlots++
+					require.True(t, slotResp.Slot.Container != nil)
+				}
+			}
+			require.Equal(t, nodeToSlots[agent.Id], activeSlots)
+		}
+	}
+
+	// At the moment, ROCm resource manager should panic when this device type is used in requests
+	// about cluster resources because ROCm is currently not supported in Kubernetes, so we check
+	// this resource manager separately.
+	require.Panics(t, testROCMResourceManager)
+}
+
+func testROCMResourceManager() {
+	auxNode := createMockNode(auxNodeName)
+	nodes := make(map[string]*k8sV1.Node)
+	nodes[auxNode.Name] = &auxNode
+	rm := createMockRM(nodes, device.ROCM)
+	rm.GetAgents(&apiv1.GetAgentsRequest{})
+}
+
+func createMockNode(name string) k8sV1.Node {
+	resourceList := createMockResourceList()
+	nodeStatus := k8sV1.NodeStatus{Allocatable: resourceList}
+	return k8sV1.Node{
+		ObjectMeta: metaV1.ObjectMeta{
+			ResourceVersion: "1",
+			Name:            name,
+		},
+		Status: nodeStatus,
+	}
+}
+
+// createMockPodsService creates two pods. One pod is run on the auxillary node and the other is
+// run on the compute node.
+func createMockPodsService(nodes map[string]*k8sV1.Node, devSlotType device.Type) *pods {
+	// Create pods.
+	pod1 := createMockPod(auxNodeName, pod1NumSlots)
+	pod2 := createMockPod(compNodeName, pod2NumSlots)
+
+	podHandlers := make(map[string]*pod)
+	podHandlers[string(pod1.allocationID)] = pod1
+	podHandlers[string(pod2.allocationID)] = pod2
```

Comment:
similar to other comments,
```suggestion
	podHandlers := map[string]*pod{
		string(pod1.allocationID)]: pod1,
		string(pod2.allocationID)]: pod2
	}
```